features/protoc/main.go

```diff
@@ -2287,3 +2301,9 @@ func newFileInfo(file *protogen.File) *fileInfo {
 
 	return f
 }
+
+// checks whether the package path is an internal package
+func isInternalPackage(pkg string) bool {
+	pathParts := strings.Split(pkg, "/")
+	return slices.Contains(pathParts, "internal")
```

Comment:
We have in the sdk `testutil/testdata/testpb` that would still be registered. Is that fine or should we skip `testutil` as well?

Reply:
I don't have a strong opinion here

Reply:
Are `testdata` packages definitely internal to other packages or go modules?

Reply:
> The go tool will ignore a directory named "testdata", making it available
to hold ancillary data needed by the tests.

I know it is ignored by `go test`, but I believe it can just be imported e.g: https://github.com/cardano-foundation/cardano-ibc-incubator/blob/main/cosmos/sidechain/app/app.go#L25

Reply:
Okay. What's your preference?

In some ways I'd be in favor of disabling all global registration but that might create some pretty unexpected behavior

Reply:
I think we should maybe keep it then, just in case indeed.

Reply:
We can always add later